src/database/migrations/0069-create-files.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE files (
+	id integer PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	name text NOT NULL,
+	storage_key uuid UNIQUE DEFAULT gen_random_uuid(),
+	mime_type text NOT NULL,
+	size integer NOT NULL CHECK (size >= 0),
+	created_by uuid NOT NULL REFERENCES users (keycloak_user_id)
+	ON DELETE CASCADE,
+	created_at timestamp with time zone NOT NULL DEFAULT now()
+);
```

Comment:
Is the URL to see the file an implicit composition of the various S3 env vars, etc.? Should we add the expected URL to the file at the time of row insert in case of S3 env var changes?

Reply:
The URL to see the file is a thing that will be generated separately, since there is no such thing as a universally-accessible-non-expiring-public-URL (that would be a security risk.

Reply:
Oh right, well, I assume no secret keys are in a URL, that would be bad ; ) ; )

If there is a secret key in a URL I suppose it would be in query parameters or after a pound sign or whatever, so I would not include those, my assumption is there is still a root URL onto which those keys get appended for temporary access.

Reply:
There are secrets in the URL for sure (that's what makes it a presigned URL) but that's also the security model.  The idea is that the temporary presigned URL provides access for the user it was generated for and is, itself, the secret.

You can read more about it here -- https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html

It doesn't make much sense to me to store a URL that can't be used as opposed to structured attributes that signal where the object is stored + allows us to generate presigned urls using the S3 protocol.

Reply:
@slifty My double wink was to no avail. I recall having this discussion in the past regarding secrets in the URL. The wink was supposed to be a reference to that.

The link you shared has pre-signed URLs starting with a URL or perhaps URI, the Location or Identifier of a Resource (a document or file), and then includes as additional parameters the credentials to access that Resource.

If I'm the object owner, can I not make a specially formed request to GET that Resource without the parameters? In other words, all the stuff up to the parameters Locates or Identifies a Resource (file) that we are interested in.